### PR TITLE
Update compatibility with fastai v2

### PR DESCRIPTION
**Pull request by lluissalord**
## What changes are proposed in this pull request?

Currently MLFlow is supporting fastai_v1 which is already deprecated. Hence, here I propose several changes which updates MLFlow to be aligned with fastai_v2

## How is this patch tested?

Unit Tests updated here

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Compatibility with last version of fastai framework.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by lluissalord** [comment]
I tested locally and it was passing the tests. Anyone here that could help me figure out why are they failing now?

**Comment by lluissalord** [comment]
The tests which are failing seem to not be related to this PR:
* **lint**: Related to spacy tests which have not been impacted on this PR.
* **flavors**: Unsure if it is related to a SqlAlchemyStore.

Please anyone that could review this PR? Do you need anything from my side (comments on code, more detailed description, etc.)?

Thank you

**Comment by lluissalord** [comment]
@smurching, I am sorry to bother you, I see that you participated on the initial implementation of Fastai flavor previously, could you please help me reviewing this PR o guide me on how should I proceed to get this PR merged (add labels, tag someone, etc.)?

Thank you in advance.

**Comment by smurching** [comment]
Hi @lluissalord, thanks for reaching out and for the PR! I unfortunately don't have bandwidth to help review at the moment, but looping in @harupy @WeichenXu123 who may be able to help :)

**Comment by harupy** [comment]
@smurching Can we drop support for fastai 1.x (I'd like to drop it because maintaining both 1.x and 2.x is a bit painful)? If yes, should we deprecate it in the next release, then drop it?

**Comment by lluissalord** [comment]
@smurching @harupy Is there anything I could do to help this to go forward?

File: mlflow/fastai/__init__.py [hunk 1]
```diff
@@ -67,22 +65,30 @@ def get_default_conda_env(include_cloudpickle=False):
...
         import cloudpickle
 
         pip_deps = ["cloudpickle=={}".format(cloudpickle.__version__)]
     return _mlflow_conda_env(
-        additional_conda_deps=["fastai={}".format(fastai.__version__)],
+        additional_conda_deps=[
+            "fastai={}".format(fastai.__version__),
+            "pytorch={}".format(torch.__version__),
+            "torchvision={}".format(torchvision.__version__),
+        ],
```
Comment (jinzhang21) [inline mlflow/fastai/__init__.py:89]: For context, MLflow is moving towards a conda-free environment (in favor of pip). Could you pull from the master branch, update these deps as pip_deps to resolve conflicts? Thanks!

**Review by jinzhang21 — COMMENTED** [review]
Thanks for the contribution for adding fastai v2 to MLflow, Lluís! I can help to review this PR going forward.

I noticed that there's a conflict of your PR with the latest master branch. Could you please resolve it to unblock tests? I'll review the rest of the PR once confirmed all tests are passing. Thanks!

**Comment by jinzhang21** [comment]
Ping. @lluissalord, do you have time in the near future to address the feedback (moving conda deps to pips)? Thanks!

**Comment by lluissalord** [comment]
@jinzhang21 I am sorry that I do not have time to make the changes on this PR. As you probably cannot modify this PR, I have invited you as collaborator of my repo, if you want to improve this PR.

**Pull request closed without merging**